lib/spack/docs/tutorial_buildsystems.rst

```diff
@@ -0,0 +1,499 @@
+.. _build-systems-tutorial:
+
+==============================
+Spack Package Build Systems
+==============================
+
+You may begin to notice after writing a couple of package template files a 
+pattern emerge for some packages. For example, you may find yourself writing
+an :code:`install` method that invokes: :code:`configure`, :code:`cmake`, 
```

Comment:
The rest of the documentation uses \`\`install\`\` instead of :code:\`install\`. It would be good to keep things consistent.